elements/lisk-p2p/test/unit/peer/inbound.ts

```diff
@@ -13,67 +13,140 @@
  *
  */
 import { expect } from 'chai';
-import { SCServerSocket } from 'socketcluster-server';
-import { InboundPeer } from '../../../src/peer';
+import { InboundPeer, PeerConfig } from '../../../src/peer';
 import { P2PDiscoveredPeerInfo } from '../../../src/p2p_types';
+import { SCServerSocket } from 'socketcluster-server';
+import { DEFAULT_RANDOM_SECRET } from '../../../src/constants';
 
 describe('peer/inbound', () => {
-	const DEFAULT_RANDOM_SECRET = 123;
-	const defaultPeerInfo: P2PDiscoveredPeerInfo = {
-		ipAddress: '12.12.12.12',
-		wsPort: 5001,
-		height: 545776,
-		isDiscoveredPeer: true,
-		version: '1.1.1',
-		protocolVersion: '1.1',
-	};
-
-	const socket = <SCServerSocket>({
-		on: sandbox.stub(),
-	} as any);
-
-	const defaultPeer = new InboundPeer(defaultPeerInfo, socket, {
-		rateCalculationInterval: 1000,
-		wsMaxMessageRate: 1000,
-		wsMaxMessageRatePenalty: 10,
-		secret: DEFAULT_RANDOM_SECRET,
-		maxPeerInfoSize: 10000,
-		maxPeerDiscoveryResponseLength: 1000,
+	let defaultPeerInfo: P2PDiscoveredPeerInfo;
+	let defaultPeerConfig: PeerConfig;
+	let defaultInboundPeer: InboundPeer;
+	let inboundSocket: SCServerSocket;
+
+	beforeEach(() => {
+		defaultPeerInfo = {
+			ipAddress: '12.12.12.12',
+			wsPort: 5001,
+			height: 545776,
+			isDiscoveredPeer: true,
+			version: '1.1.1',
+			protocolVersion: '1.1',
+		};
+		defaultPeerConfig = {
+			rateCalculationInterval: 1000,
+			wsMaxMessageRate: 1000,
+			wsMaxMessageRatePenalty: 10,
+			secret: DEFAULT_RANDOM_SECRET,
+			maxPeerInfoSize: 10000,
+			maxPeerDiscoveryResponseLength: 1000,
+		};
+		inboundSocket = <SCServerSocket>({
+			on: sandbox.stub(),
+			off: sandbox.stub(),
+			emit: sandbox.stub(),
+			destroy: sandbox.stub(),
+		} as any);
+		sandbox.spy(global, 'setTimeout');
+		defaultInboundPeer = new InboundPeer(
+			defaultPeerInfo,
+			inboundSocket,
+			defaultPeerConfig,
+		);
+	});
+
+	afterEach(() => {
+		defaultInboundPeer.disconnect();
 	});
 
 	describe('#constructor', () => {
-		it('should be an object', () => {
-			return expect(defaultPeer).to.be.an('object');
-		});
+		it('should be an instance of InboundPeer class', () =>
+			expect(defaultInboundPeer).and.be.instanceof(InboundPeer));
+
+		it('should have a function named _handleInboundSocketError', () =>
+			expect((defaultInboundPeer as any)._handleInboundSocketError).to.be.a(
+				'function',
+			));
+
+		it('should have a function named _handleInboundSocketClose ', () =>
+			expect((defaultInboundPeer as any)._handleInboundSocketClose).to.be.a(
+				'function',
+			));
+
+		it('should should have a function named _sendPing', () =>
+			expect((defaultInboundPeer as any)._sendPing).to.be.a('function'));
 
-		it('should be an instance of P2P blockchain', () => {
-			return expect(defaultPeer)
-				.to.be.an('object')
-				.and.be.instanceof(InboundPeer);
+		it('should set ping timeout', () => {
```

Comment:
`setPing` has a callback, shouldn't it be `async`?

Reply:
Ah, you can ignore this comment, I thought there is an async task within `setPing` function